### PR TITLE
fix(core): Fix oauth2 callback and add integration tests (no-changelog)

### PR DESCRIPTION
**Pull request by netroy**
## Summary
Accidentally broke oauth callback in https://github.com/n8n-io/n8n/pull/10263 .
To prevent that from happening again, this PR also adds integration tests.

## Review / Merge checklist

- [x] PR title and summary are descriptive
- [x] Tests included

### CODE REVIEW

File: packages/cli/src/controllers/oauth/oAuth1Credential.controller.ts [hunk 1]
```diff
@@ -33,7 +33,7 @@ export class OAuth1CredentialController extends AbstractOAuthController {
 	@Get('/auth')
 	async getAuthUri(req: OAuthRequest.OAuth1Credential.Auth): Promise<string> {
```
Comment (tomi) [inline packages/cli/src/controllers/oauth/oAuth1Credential.controller.ts:34]: Is the type here correct if `req.user` was not caught by the type system? Same question to the other endpoint handlers

File: packages/cli/test/integration/controllers/oauth/oauth2.api.test.ts [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+import { CredentialsHelper } from '@/CredentialsHelper';
+import { OAuth2CredentialController } from '@/controllers/oauth/oAuth2Credential.controller';
+
+import { createOwner } from '@test-integration/db/users';
+import { saveCredential } from '@test-integration/db/credentials';
+import * as testDb from '@test-integration/testDb';
+import { setupTestServer } from '@test-integration/utils';
+import type { SuperAgentTest } from '@test-integration/types';
+
+describe('OAuth2 API', () => {
```
Comment (tomi) [inline packages/cli/test/integration/controllers/oauth/oauth2.api.test.ts:17]: Do we need tests for Oauth1 as well? Or is the support for that deprecated?

**Review by tomi — DISMISSED** [review]
LGTM, nice tests 👏 Couple questions but can also be merged as is

File: packages/cli/src/controllers/oauth/oAuth1Credential.controller.ts (hunk 1 above)
Comment (netroy) [reply packages/cli/src/controllers/oauth/oAuth1Credential.controller.ts:34]: I think I do not understandd how `never` works. 
`AuthlessRequest` had `user: never` and somehow it was not a type error.
I've removed `user` from `AuthlessRequest`, and now `req.user` is a type-error in this context.

Screenshot from 2024-07-31 16-29-49

File: packages/cli/test/integration/controllers/oauth/oauth2.api.test.ts (hunk 2 above)
Comment (netroy) [reply packages/cli/test/integration/controllers/oauth/oauth2.api.test.ts:17]: TBH, everytime I've asked about oauth1, I've only heard that some folks might be using it for Twitter, but I haven't actually seen anyone use it. 
In the past we've had bugs in oauth1 flow, and no one reported them either.
In the last few months, we've refactored this code to make oauth1 flow use a lot more code shared with oauth2 flow. 
So technically this already covers quite a bit of the oauth1 flow as well.
But, I don't think it's worth the effort to write dedicated integration tests for oauth1.
We can add these tests when someone reports a bug in that code.

**Review by tomi — APPROVED** [review]

**Comment by janober** [comment]
Got released with [`n8n@1.54.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.54.0)

**Pull request merged**